joomla/joomla-cms
# [com_fields] Handling single 0 selections properly in checkboxes

Pull Request for Issue #12810.

### Summary of Changes
Adds a callback functionality for form fields to prepare the value of a custom field before the value is passed to the form.

### Testing Instructions
- Create a custom field of the type checkboxes and assign two options, one of them with a 0 value. Eg: 
![image](https://cloud.githubusercontent.com/assets/251072/20260205/e504d154-aa58-11e6-97c9-a603d718f3c5.png).
- Select only the 0 value ("No" in my example) and save.
- Reopen the form and check the value.
- Check the item in frontend.

### Expected result
When editing the article the 0 checkbox should be selected. On the frontend "No" should be displayed.

### Actual result
No checkbox is selected in the form and nothing is displayed when viewing an article.
